### PR TITLE
refactor(android)(game): architecture unification and UI turn control

### LINKED ISSUE #493: feat(android)(game): add local current user identity source
## Description

Describe the feature briefly.

## Acceptance Criteria

- [ ]

## Notes

Additional context, screenshots, or links.

### PR DESCRIPTION
**Pull request by ez-del**
## Summary
This PR refactors the Android game screen architecture to make UI interaction and turn-based control more consistent and easier to maintain.

It introduces a clearer `GameUIEvent` / `GameEffect` flow, centralizes active-player state handling, updates navigation from `matchId` to `gameId`, and rewires the screen/components to use event-driven interaction instead of passing raw ViewModel actions through the UI. It also updates the related tests accordingly.

## Issues

- Closes #519
- Closes #507
- Closes #502 
- Closes #496
- Closes #494
- Closes #493
- Closes #483

## What changed
- introduced `GameUIEvent` / `GameEffect` to make screen interaction more explicit and structured
- updated game screen components to dispatch events instead of directly depending on raw ViewModel passthrough callbacks
- added / derived active-turn state (`isActivePlayer`) to better control which actions are available to the current user
- integrated the new event/effect handling into the `GameScreen` and `GameViewModel`
- updated navigation to use `gameId` instead of `matchId`
- updated affected Android tests and added coverage-related follow-up fixes.

---

The goal of this refactor is to reduce ad-hoc screen wiring and move the game UI toward a more unified architecture:
- clearer separation between UI events, side effects, and state
- cleaner control of turn-based actions
- less direct coupling between composables and ViewModel internals
- easier future extension for loading, error, and multiplayer turn-state handling.

## Result
After this PR, the Android game screen follows a more consistent event-driven structure, active-player restrictions are handled more centrally, and the game route / screen integration is better aligned with the underlying game model.

### CODE REVIEW

File: apps/android/app/src/main/java/at/aau/serg/android/ui/screens/game/GameScreen.kt [hunk 1]
```diff
@@ -28,18 +28,27 @@ import at.aau.serg.android.ui.theme.ThemeState
...
+    val uiState by viewModel.uiState.collectAsState()
+
+    GameScreenContent(
+        uiState = uiState,
+        onEvent = viewModel::onUIEvent
+    )
+}
+
+@Composable
+fun GameScreenContent(
```
Comment (kingjulien1) [inline apps/android/app/src/main/java/at/aau/serg/android/ui/screens/game/GameScreen.kt:52]: I think this PR is a good opportunity to split `GameScreenContent` a bit more aggressively.

Now that the screen is already moving toward a cleaner `GameUIEvent`/`GameEffect` architecture, I’d extract the main sections (for example header/actions, board area, rack area) into smaller composables and keep `GameScreenContent` focused on wiring high-level state + callbacks together.

Sonar is already flagging the complexity here, so this would also reduce that pressure in a meaningful way.

File: apps/android/app/src/main/java/at/aau/serg/android/ui/screens/game/GameUIEvent.kt [hunk 2]
```diff
@@ -0,0 +1,24 @@
+package at.aau.serg.android.ui.screens.game
+
+import shared.models.game.domain.Tile
+
+sealed interface GameUIEvent {
+    val requiresActivePlayer: Boolean get() = true
```
Comment (kingjulien1) [inline apps/android/app/src/main/java/at/aau/serg/android/ui/screens/game/GameUIEvent.kt:6]: I like this direction a lot — `requiresActivePlayer` makes the UI-event model much clearer.

I’d just make sure this contract is really enforced in one central place (presumably in `GameViewModel.onUIEvent(...)`) and covered by tests. Otherwise it is easy for this to become a “documentation property” that looks nice in the sealed interface but is not actually guaranteed by behaviour.

File: apps/android/app/src/main/java/at/aau/serg/android/ui/navigation/HomeGraph.kt [hunk 3]
```diff
@@ -124,18 +126,31 @@ fun NavGraphBuilder.homeGraph(
             AuthScreen(viewModel = vm)
         }
 
-        composable("${Routes.GAME}/{matchId}") {
+        composable("${Routes.GAME}/{gameId}") {
+            val gameId = it.arguments?.getString("gameId")!!
```
Comment (kingjulien1) [inline apps/android/app/src/main/java/at/aau/serg/android/ui/navigation/HomeGraph.kt:130]: The rename from `matchId` to `gameId` definitely looks like the right direction here.

I’d just double-check that all navigation call sites, deeplink assumptions, and any tests/builders that still construct the old route string were updated consistently as well. This kind of route-argument rename is easy to miss in one or two places and then only shows up later as a navigation/runtime bug.

File: apps/android/app/src/main/java/at/aau/serg/android/ui/screens/game/GameUiState.kt [hunk 4]
```diff
@@ -16,4 +16,7 @@ data class GameUiState(
     val activeSelectionRow: String? = null,
     val gameState: ConfirmedGame? = null,
     val winnerUserId: String? = null
-)
+) {
+    val isActivePlayer: Boolean
+        get() = user != null && gameState?.currentPlayerUserId == user.uid
```
Comment (kingjulien1) [inline apps/android/app/src/main/java/at/aau/serg/android/ui/screens/game/GameUiState.kt]: I like having this as a derived UI-state property instead of recomputing it in multiple places.

I’d just make sure the tests explicitly cover the edge cases here too: `user == null`, `gameState == null`, and a loaded game where the active player changes underneath the current user. Since several controls are now gated off this one property, it’s worth locking those cases down directly.

**Review by kingjulien1 — CHANGES_REQUESTED** [review]
Nice refactor overall 👍 — the UI-event/effect direction looks much cleaner, and I only left a few smaller comments around complexity, central contract enforcement, and making sure the route rename is fully consistent.

Please feel more than free to respond to my review comments if you have different opinions / resolution ideas or completely disagree.

File: apps/android/app/src/main/java/at/aau/serg/android/ui/screens/game/GameUIEvent.kt (hunk 2 above)
Comment (ez-del) [reply apps/android/app/src/main/java/at/aau/serg/android/ui/screens/game/GameUIEvent.kt:6]: That's exactly how this is done.

File: apps/android/app/src/main/java/at/aau/serg/android/ui/navigation/HomeGraph.kt (hunk 3 above)
Comment (ez-del) [reply apps/android/app/src/main/java/at/aau/serg/android/ui/navigation/HomeGraph.kt:130]: That is a pretty much a string comaprison so it matters little how this is called how other screens pass that.

File: apps/android/app/src/main/java/at/aau/serg/android/ui/screens/game/GameUiState.kt (hunk 4 above)
Comment (ez-del) [reply apps/android/app/src/main/java/at/aau/serg/android/ui/screens/game/GameUiState.kt]: Has been changed, this however introduces a very dangerous option of having IsActivePlayer stale which would allow actions which are not allowed or should be allowed because of incorrect implementation later on.

**Review by kingjulien1 — DISMISSED** [review]

**Review by kingjulien1 — APPROVED** [review]
